### ISSUE TITLE
Treat alias names like description

### ISSUE THREAD

**Issue by nomeata** [NixOS/nixos-search#345]
I was searching for `manpages` (the package name I was used from Debian), but https://search.nixos.org/packages?channel=21.05&from=0&size=50&sort=relevance&query=manpages did not show anything. Then someone told me that `manpages` exists, leaving me confused. Finally someone explained that the search hides aliases.

Hiding alias from the results is sensible, but it would probably still make sense to use that data when searching, i.e. show `man-pages` when someone searches for `manpages`, as if `manpages` appeared in the description.

**Comment by ulysses4ever** [comment]
I confirm, as a novel Nix user, that I ran into the searching by alias problem multiple times and heard from people who also had that. The web search serves novices in large part (I imagine), so it should strive to make any usability improvement it can (I.e. saying: just use `nix search` doesn’t really suffice). 

I understand that there’s a technical challenge here, so… Just a user report that it would help if there was a volunteer who implements it.

**Comment by jtojnar** [comment]
I would say it would be better to just improve package descriptions, rather than try to do anything with legacy aliases that will eventually get removed. Or add tags to packages.

**Comment by nomeata** [comment]
Good point. Maybe we can simply add useful aliases to the package descriptions manually? Throw in a simple took that looks at `aliases.nix` and reports aliases that don't contain ther name in the aliases? And, if necessary, add an “extra index words” list to the package meta to add them without artistically changing a good description.

This might be easier to pull off, and yield better results

**Comment by ncfavier** [comment]
Another problem is that, even ignoring aliases, nix-env will not include syntactically duplicate derivations (https://github.com/NixOS/nix/commit/4eb637c799bddfcff7661164f98cd604fe4120f5), so e.g. `linux_latest` is not included.

I'm starting to think we should stop relying on nix-env to get packages from nixpkgs and just treat nixpkgs like any other flake... @ysndr thoughts?

**Comment by ysndr** [comment]
That was the plan initially, but was blocked since nixpkgs is based on legacyPackages which are 

a) not evaluated/listed by `nix flake show`
b) troublesome to evaluate by hand (all the `nixpkgs_allow_*` stuff)

\* Support for `legacyPackages` is actually present already (maybe just not exposed)  

Maybe your recent changes in nix changed the situation a bit (i dont keep up too closely with nix).

Last time avoiding evaluation was only possible if nix had a more thorough `--json` flag for `nix flake show` and `nix flake show --legacy`.
However, i don't know if that would change the alias situation.

**Comment by ncfavier** [comment]
This came up again: https://github.com/NixOS/nixos-search/issues/563
